packages/@aws-cdk/pipelines/lib/stage.ts

```diff
@@ -76,14 +81,79 @@ export class CdkStage extends CoreConstruct {
    */
   public addApplication(appStage: Stage, options: AddStageOptions = {}) {
     const asm = appStage.synth({ validateOnSynthesis: true });
-    const extraRunOrderSpace = options.extraRunOrderSpace ?? 0;
+    const extraRunOrderSpace = options.extraRunOrderSpace ?? 0 + (options.securityCheck ? 2 : 0);
 
     if (asm.stacks.length === 0) {
       // If we don't check here, a more puzzling "stage contains no actions"
       // error will be thrown come deployment time.
       throw new Error(`The given Stage construct ('${appStage.node.path}') should contain at least one Stack`);
     }
 
+    if (options.securityCheck) {
+      const preApproveLambda = new lambda.Function(this, `${this.stageName}AutoApprove`, {
+        functionName: 'AutoApproveLambda',
+        handler: 'index.handler',
+        runtime: lambda.Runtime.PYTHON_3_8,
+        code: lambda.Code.fromAsset(resolve(__dirname, 'lambda')),
+      });
+
+      const invokeLambda =
+        'aws lambda invoke' +
+        ' --function-name AutoApproveLambda' +
+        ' --invocation-type Event' +
+        ' --payload "$payload"' +
+        ' lambda.out';
+
+      const build = new codebuild.Project(this, 'CDKSecurityCheck', {
+        buildSpec: codebuild.BuildSpec.fromObject({
+          version: 0.2,
+          phases: {
+            build: {
+              commands: [
+                'npm install -g aws-cdk',
+                // $CODEBUILD_INITIATOR will always be Code Pipeline and in the form of:
+                // "codepipeline/example-pipeline-name-Xxx"
+                // cut splits the string based on the delimiter "/" and returns the 2nd field
+                'payload="$(jq -n --arg pipelineName "$(echo $CODEBUILD_INITIATOR | cut -d "/" -f 2 )" --arg stageName "$STAGE_NAME" \'{ PipelineName: $pipelineName, StageName: $stageName }\' )"',
+                'cdk diff -a . --security-only --fail',
+                `$? || ${invokeLambda}`,
```

Comment:
Why does this work? This will evaluate to:

```
1 || aws lambda invoke-function ...

or 

0 || aws lambda invoke-function ...
```

Does the literal string `0` or `1` lead to its exit code?

I would have expected this all on one line:

```
cdk dif ... || ${invokeLambda}
```

Reply:
I did this before and reverting it back in the next commit.. was doing some testing 🙃